### DISCUSSION TITLE
TNT

### DISCUSSION THREAD

**Discussion by opekope2** [opekope2/StaffMod#25] [Staff blocks]
TNT can be seen at multiple times in the video. Green wields it at [2:07-2:12](https://youtu.be/vlVUVsx6BYY?t=127). The TNTs don't fly very fast, and explode when hit the ground.

Attack speed: same as empty staff
Attack damage: same as empty staff

Planned implementation: left click to shoot a TNT, which explodes on collision, including collision with entities. If it collides with another "impact TNT", both of them explode. Left-clicking prevents TNT spam without moving or cheats, because the player would attack the TNT instead (if it's too close)

Other features to discuss:
* Ignore other TNT entities when checking for collisions
* When attacking, spawn a weak explosion at the victim's position (and maybe yeet the victim into the air)
* Right click to explode the last thrown TNT
* Excavation TNT: explode on impact, but don't despawn until its fuse runs out (when it explodes once more and despawns)

**Comment by opekope2** [comment]
Yet another feature idea, when I implement [advanced block functionality](https://github.com/opekope2/StaffMod/discussions/6#discussioncomment-8162782): configurable fuse length. This may be used to troll an unsuspecting player to shoot a TNT only to explode 1 tick later, for example

**Reply by charmander9695** [reply]
that could be useful in some ways P.s. sorry for not talking in a while
